internal/providers/terraform/aws/autoscaling_group.go

```diff
@@ -48,227 +44,152 @@ func NewAutoscalingGroup(d *schema.ResourceData, u *schema.UsageData) *schema.Re
 	mixedInstanceLaunchTemplateRef := d.References("mixed_instances_policy.0.launch_template.0.launch_template_specification.0.launch_template_id")
 
 	if len(launchConfigurationRef) > 0 {
-		lc := newLaunchConfiguration(launchConfigurationRef[0].Address, launchConfigurationRef[0], u, region)
-
-		// AutoscalingGroup should show as not supported LaunchConfiguration is not supported
-		if lc == nil {
-			return nil
-		}
-		schema.MultiplyQuantities(lc, desiredCapacity)
-		subResources = append(subResources, lc)
+		data := launchConfigurationRef[0]
+		a.LaunchConfiguration = newLaunchConfiguration(data, u, a.Region, instanceCount)
 	} else if len(launchTemplateRef) > 0 {
-		onDemandCount := desiredCapacity
-		spotCount := decimal.Zero
-		if strings.ToLower(launchTemplateRef[0].Get("instance_market_options.0.market_type").String()) == "spot" {
-			onDemandCount = decimal.Zero
-			spotCount = desiredCapacity
-		}
-		lt := newLaunchTemplate(launchTemplateRef[0].Address, launchTemplateRef[0], u, region, onDemandCount, spotCount)
+		data := launchTemplateRef[0]
 
-		// AutoscalingGroup should show as not supported LaunchTemplate is not supported
-		if lt == nil {
-			return nil
+		onDemandPercentageAboveBaseCount := int64(100)
+		if strings.ToLower(data.Get("instance_market_options.0.market_type").String()) == "spot" {
+			onDemandPercentageAboveBaseCount = int64(0)
 		}
-		subResources = append(subResources, lt)
-	} else if len(mixedInstanceLaunchTemplateRef) > 0 {
-		mixedInstancesPolicy := d.Get("mixed_instances_policy.0")
-		lt := newMixedInstancesAwsLaunchTemplate(mixedInstanceLaunchTemplateRef[0].Address, mixedInstanceLaunchTemplateRef[0], u, region, desiredCapacity, mixedInstancesPolicy)
 
-		// AutoscalingGroup should show as not supported LaunchTemplate is not supported
-		if lt == nil {
-			return nil
-		}
-		subResources = append(subResources, lt)
-	}
-
-	return &schema.Resource{
-		Name:         d.Address,
-		SubResources: subResources,
-	}
-}
-
-func newLaunchConfiguration(name string, d *schema.ResourceData, u *schema.UsageData, region string) *schema.Resource {
-	tenancy := "Shared"
-	if strings.ToLower(d.Get("placement_tenancy").String()) == "host" {
-		log.Warnf("Skipping resource %s. Infracost currently does not support host tenancy for AWS Launch Configurations", d.Address)
-		return nil
-	} else if strings.ToLower(d.Get("placement_tenancy").String()) == "dedicated" {
-		tenancy = "Dedicated"
+		a.LaunchTemplate = newLaunchTemplate(data, u, a.Region, instanceCount, int64(0), onDemandPercentageAboveBaseCount)
+	} else if len(mixedInstanceLaunchTemplateRef) > 0 {
+		data := mixedInstanceLaunchTemplateRef[0]
+		a.LaunchTemplate = newMixedInstancesLaunchTemplate(data, u, a.Region, instanceCount, d.Get("mixed_instances_policy.0"))
 	}
 
-	instanceType := d.Get("instance_type").String()
+	a.PopulateUsage(u)
```

Comment:
I think `u` is usageData for the autoscaling group right?  Is that ok or should we be passing in the usageData associated with the referenced LaunchConfiguration?

Reply:
Looking at the existing tests we set the usage on the ASG, e.g.:
```yml
  aws_autoscaling_group.asg_lc_reserved:
    reserved_instance_type: standard
    reserved_instance_term: 1_year
    reserved_instance_payment_option: no_upfront
```
So this works, but it won't generate the empty usage file correctly. I think I should add these as usage parameters on the ASG and pass them down to the launch template/configuration.

Reply:
Pushed a fix that will generate the empty usage file correctly.

Reply:
Ok that makes sense.  I guess we should think about adding golden files for --sync-usage-file output.